src/virtual_machine_start.rs

```diff
@@ -0,0 +1,120 @@
+                writeln!(&stream, "{}", response).map_err(|error| error.to_string())?;
+            }
+            "r" => {
+                let response = virtual_machine.state_of_registers();
+                writeln!(&stream, "{}", response).map_err(|error| error.to_string())?;
+            }
+            _ => match command.parse::<u16>() {
+                Ok(amount_of_instructions) => {
+                    virtual_machine.next_instructions(amount_of_instructions);
+                    let response = format!("executed {} instructions", amount_of_instructions);
+                    writeln!(&stream, "{}", response).map_err(|error| error.to_string())?;
+                }
+                Err(_) => {
+                    writeln!(&stream, "Invalid Command").map_err(|error| error.to_string())?
+                }
+            },
+        }
+    }
+}
+
+fn print_instructions_for_interactive_console() {
+    println!("Starting interactive console");
+    println!("Instructions: ");
+    println!("<r> to print the state of the registers at the moment");
+    println!("<an instruction in binary> to instantly execute that instruction");
+}
+
+pub fn execute_vm_in_interactive_mode() -> Result<(), String> {
+    let program_counter_start = 0x3000;
+    let mut virtual_machine = LC3VirtualMachine::new(program_counter_start);
+    print_instructions_for_interactive_console();
+    for line in stdin().lock().lines() {
+        let line = line.map_err(|error| error.to_string())?;
+        match line.as_str() {
+            "r" => {
+                let registers = virtual_machine.state_of_registers();
+                let registers = registers.split(STREAM_DATA_SEPARATOR);
+                for register in registers {
+                    println!("{}", register);
+                }
+            }
+            _ => match u16::from_str_radix(&line, 2) {
+                Ok(instruction) => {
+                    virtual_machine.decode_instruction(instruction);
+                    let instruction = format!("{instruction:#018b}",);
+                    println!("instruction proccess: {}", instruction);
+                }
+                Err(_) => println!("Wrong instruction format"),
+            },
+        }
+    }
+    Ok(())
+}
+
+fn receive_file(path: impl AsRef<Path>) -> Result<BufReader<File>, String> {
+    let file = File::open(path).map_err(|error| error.to_string())?;
+    let file_reader = BufReader::new(file);
+    Ok(file_reader)
+}
+
+fn load_reader_file_to_vm_memory(mut reader: BufReader<File>) -> Result<LC3VirtualMachine, String> {
+    let program_counter_start = reader
+        .read_u16::<BigEndian>()
+        .map_err(|error| error.to_string())?;
+
+    let mut virtual_machine = LC3VirtualMachine::new(program_counter_start);
+    let mut memory_address = program_counter_start;
+    while let Ok(instruction) = reader.read_u16::<BigEndian>() {
+        virtual_machine.memory_write(memory_address, instruction);
+        memory_address += 1;
```

Comment:
```suggestion
        memory_address = memory_address.wrapping_add(1);
```
To prevent `u16` overflow.

Reply:
Oops, missed that one